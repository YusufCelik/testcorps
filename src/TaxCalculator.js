import React from "react";
import Button from "./Button";

class TaxCalculator extends React.Component {
  constructor() {
    super();
    this.revenueInput = React.createRef();

    this.state = {
      total: 0
    };
  }

  changeLabel() {
    this.setState({
      total: this.revenueInput.value
    });
  }

  render() {
    return (
      <section>
        <input
          data-testhook-id="tax-calculator-revenue-input"
          type="number"
          ref={ref => (this.revenueInput = ref)}
        />
        <Button label="shmalculate" onClick={() => this.changeLabel()} />
        <p data-testhook-id="tax-calculator-total">{this.state.total}</p>
      </section>
    );
  }
}

export default TaxCalculator;
