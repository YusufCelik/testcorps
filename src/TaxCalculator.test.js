import React from "react";
import { shallow, mount } from "enzyme";

import TaxCalculator from "./TaxCalculator";
import Button from "./Button";

describe("<TaxCalculator />", () => {
  it("renders calculate button", () => {
    Date.now = jest.fn(() => 1482363367071);
    const wrapper = shallow(<TaxCalculator />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders tax total label", () => {
    const wrapper = shallow(<TaxCalculator />);
    expect(
      wrapper.find('[data-testhook-id="tax-calculator-total"]').exists()
    ).toBe(true);
  });

  it("changes tax total when user clicks calculate", () => {
    const wrapper = mount(<TaxCalculator />);
    wrapper
      .find('[data-testhook-id="tax-calculator-revenue-input"]')
      .instance().value = "20";

    wrapper
      .find('[data-testhook-id="tax-calculator-revenue-input"]')
      .simulate("change");

    wrapper.find(Button).simulate("click");
    expect(
      wrapper.find('[data-testhook-id="tax-calculator-total"]').text()
    ).toEqual("20");

    expect(wrapper.instance().state).toEqual({ total: "20" });
    wrapper.setState({ total: "40" });

    expect(
      wrapper.find('[data-testhook-id="tax-calculator-total"]').text()
    ).toEqual("40");
  });
});
