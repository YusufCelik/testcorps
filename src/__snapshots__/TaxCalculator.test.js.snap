// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaxCalculator /> renders calculate button 1`] = `
<section>
  <p>
    1482363367071
  </p>
  <input
    data-testhook-id="tax-calculator-revenue-input"
    type="number"
  />
  <Button
    label="calculate"
    onClick={[Function]}
  />
  <p
    data-testhook-id="tax-calculator-total"
  >
    0
  </p>
</section>
`;
